splash/qtrender_lua.py

```diff
@@ -456,10 +456,13 @@ def get_viewport_size(self):
     @command()
     def set_viewport_size(self, width, height):
         self.tab.set_viewport('%dx%d' % (width, height))
+        process_pending_events()
 
     @command(multiple_return_values=True)
     def set_viewport_full(self):
-        return list(self.tab.set_viewport('full'))
+        size = self.tab.set_viewport('full')
+        process_pending_events()
```

Comment:
technically, it could, if it has been scheduled before the control was lended over to Lua script. or if an event that would cause a redirect has been received before that.